draft-gpew-priv-ppm.md

```diff
@@ -1004,21 +1125,139 @@ fetch certificates: https://datatracker.ietf.org/doc/html/rfc8555#section-7.4.2]
+to the collector's collect response. To do this, the leader first computes a
+checksum over the set of output shares included in the batch window identified
+by the collect request. The checksum is computed by taking the SHA256 hash of
+each nonce from the client reports included in the aggregation, then combining
+the hash values with a bitwise-XOR operation.
+
+Then, for each aggregator endpoint `[aggregator]` in the parameters associated
+with `CollectReq.task_id` (see {{collect-flow}}) except its own, the leader sends
+a POST request to `[aggregator]/aggregate_share` with the following message:
+
+~~~
+struct {
+  TaskID task_id;
+  Interval batch_interval;
+  uint64 report_count;
+  opaque checksum[32];
+  opaque helper_state<0..2^16>;
+} AggregateShareReq;
+~~~
+
+* `task_id` is the task ID associated with the PPM parameters.
+* `batch_interval` is the batch interval of the request.
+* `report_count` is the number of reports included in the aggregation.
+* `checksum` is the checksum computed over the set of client reports.
+* `helper_state` is the helper's state, which is carried across requests from
+  the leader.
+
+To handle the leader's request, the helper first ensures that the request meets
+the requirements for batch parameters following the procedure in
+{{batch-parameter-validation}}. If the batch parameters are invalid, then it
+MUST abort with error "batchInvalid".
+
+Next, it computes a checksum based on its view of the output shares included in
+the batch window, and checks that the `report_count` and `checksum` included in
+the request match its computed values. If not, then it MUST abort with error
+"batchMismatch".
+
+Next, it computes the aggregate share `agg_share` corresponding to the set of
+output shares, denoted `out_shares`, for the batch interval, as follows:
+
+~~~
+agg_share = VDAF.out_shares_to_agg_share(agg_param, out_shares)
+~~~
+
+Note that for most VDAFs, it is possible to aggregate output shares as they
+arrive rather than wait until the batch is collected. To do so however, it is
+necessary to enforce the batch parameters as described in
+{{batch-parameter-validation}} so that the aggregator knows which aggregate
+share to update.
+
+The helper then encrypts `agg_share` under the collector's HPKE public
+key as described in {{aggregate-share-encrypt}}, yielding `encrypted_agg_share`.
+Encryption prevents the leader from learning the actual result, as it only has
+its own aggregate share and cannot compute the helper's.
+
+The helper responds to the leader with HTTP status 200 OK and a body consisting
+of an `AggregateShareResp`:
+
+~~~
+struct {
+    HpkeCiphertext encrypted_aggregate_share;
+} AggregateShareResp;
+~~~
+
+`encrypted_aggregate_share.config_id` is set to the collector's HPKE config ID.
+`encrypted_aggregate_share.enc` is set to the encapsulated HPKE context `enc`
+computed above and `encrypted_aggregate_share.ciphertext` is the ciphertext
+`encrypted_agg_share` computed above.
```

Comment:
I think this discussion of how to populate the fields of `encrypted_aggregate_share` would be more clear in the `aggregate-share-encrypt` section, below.